src/server/main_service.cc

```diff
@@ -2201,6 +2216,18 @@ void Service::Pubsub(CmdArgList args, ConnectionContext* cntx) {
     PubsubChannels(pattern, cntx);
   } else if (subcmd == "NUMPAT") {
     PubsubPatterns(cntx);
+  } else if (subcmd == "NUMSUB") {
+    if (args.size() == 1) {
+      (*cntx)->SendEmptyArray();
+      return;
+    }
+
+    int channels_size = args.size() - 1;
+    std::vector<string_view> channels(channels_size);
+    for (auto i = 0; i < channels_size; i++) {
+      channels[i] = ArgS(args, i + 1);
+    }
+    PubsubNumSub(channels, cntx);
```

Comment:
To clarify, the logic of `PubsubNumSun()` should be here instead of in `ReplyBuilder`

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/0abca50381ec31ee7332cc95617af1ec9dad1eae